python/tests/unit/arcticdb/version_store/test_resample.py

```diff
@@ -9,12 +9,34 @@
 import numpy as np
 import pandas as pd
 import pytest
+import datetime as dt
 
 from arcticdb import QueryBuilder
 from arcticdb.exceptions import ArcticDbNotYetImplemented, SchemaException
 from arcticdb.util.test import assert_frame_equal
 from arcticdb.util._versions import IS_PANDAS_TWO
 
+# Pandas recommended way to resample and exclude buckets with no index values, which is our behaviour
+# See https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#sparse-resampling
+def round(t, freq):
+    freq = pd.tseries.frequencies.to_offset(freq)
+    td = pd.Timedelta(freq)
+    return pd.Timestamp((t.value // td.value) * td.value)
+
+def generic_sparse_resample_test(lib, sym, rule, aggregations, date_range=None):
```

Comment:
`sparse` might get confused with our underlying data being sparse, I'd rename this to something about empty buckets